### ISSUE TITLE
bug: onCompositionStart, onCompositionEnd, and onCompositionUpdate are bound to the wrong case-sensitive event name

### ISSUE THREAD

**Issue by sean-perkins** [ionic-team/stencil#3235]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.13.0

### Current Behavior

Currently Stencil's JSX bindings for the composition events `compositionstart`, `compositionupdate` and `compositionend` do not correctly register an event listener to the correct event name.

This appears to be an identical issue as Preact: https://github.com/preactjs/preact/issues/3003

### Expected Behavior

The event handlers for the composition events should fire when performing a composition event. 

### Steps to Reproduce

1. Create a Stencil component with an input inside the template.
2. Bind an event listener for a composition event:
```tsx
render() {
  return (
    <input 
      onCompositionStart={() => console.log('composition start')}
      onCompositionEnd={() => console.log('composition end')} />
  );
}
```
3. Perform a composition event (for instance on Mac OS - press "option" + "`", then "a"
4. Notice none of the composition events fire
5. Manually bind event listeners to the input
6. Notice the composition events fire

HTML comparison: https://developer.mozilla.org/en-US/docs/Web/API/Element/compositionstart_event

### Code Reproduction URL

https://github.com/sean-perkins/stencil-composition-events-issue

**Comment by rwaskiewicz** [comment]
Thanks! Confirmed the issue with the reproduction case and am gonna get this labeled to be ingested

**Comment by GuoJikun** [comment]
now it can use ?

**Comment by alicewriteswrongs** [comment]
Hello @GuoJikun! As you might notice above a PR addressing this issue was opened and merged into our v3 branch, so the fix will be released when Stencil v3 is released, but it won't be available on mainline releases before then.

When the fix is released we'll likely ping this issue to make sure it addresses the issue.

**Comment by rwaskiewicz** [comment]
This fix went live in Stencil v3.0. I'm not entirely sure how I missed closing this issue out as a part of the release follow up, that's on me. Apologies for that all! 

I'm gonna close this out. Should it reappear in Stencil v3.0+, please feel free to open a new issue
